WWW/smoorthi-dev/minidata_mutation_calling_v2.wdl

```diff
@@ -0,0 +1,376 @@
+version 1.0
+## Notes: Testing alignment for 3 samples
+## 
+## Samples:
+## MOLM13: Normal sample
+## CALU1: KRAS G12C mutant
+## HCC4006: EGFR Ex19 deletion mutant 
+##
+## Input requirements:
+## - combined fastq files for chromosome 12 and 7 +/- 200bp around the sites of mutation only
+##
+## Output Files:
+## - An aligned bam for all 3 samples with mutation calling
+## 
+## Workflow developed by Sitapriya Moorthi @ Fred Hutch LMD: 01/10/24 for use by DaSL @ Fred Hutch.
+## Workflow updated on : 02/13/24 by Sitapriya Moorthi
+
+struct referenceGenome {
+    File ref_fasta
+    File ref_fasta_index
+    File ref_dict
+    File ref_amb
+    File ref_ann
+    File ref_bwt
+    File ref_pac
+    File ref_sa
+    String ref_name
+}
+
+
+workflow minidata_mutation_calling_v2 {
+  input {
+    Array[File] allSamples
+    File normalFastq
+
+    referenceGenome refGenome
+    
+    File dbSNP_vcf
+    File dbSNP_vcf_index
+    File known_indels_sites_VCFs
+    File known_indels_sites_indices
+    
+    File af_only_gnomad
+    File af_only_gnomad_index
+    
+    File annovarTAR
+    String annovar_protocols
+    String annovar_operation
+  }
+ 
+  # Scatter for "tumor" samples   
+  scatter (sampleFastq in allSamples) {
+    call BwaMem as sampleBwaMem {
+      input:
+        input_fastq = sampleFastq,
+        refGenome = refGenome
+    }
+    
+    call MarkDuplicates as sampleMarkDuplicates {
+      input:
+        input_bam = sampleBwaMem.analysisReadySorted
+    }
+
+    call ApplyBaseRecalibrator as sampleApplyBaseRecalibrator{
+      input:
+        input_bam = sampleMarkDuplicates.markDuplicates_bam,
+        input_bam_index = sampleMarkDuplicates.markDuplicates_bai,
+        dbSNP_vcf = dbSNP_vcf,
+        dbSNP_vcf_index = dbSNP_vcf_index,
+        known_indels_sites_VCFs = known_indels_sites_VCFs,
+        known_indels_sites_indices = known_indels_sites_indices,
+        refGenome = refGenome
+      }
+
+    call Mutect2 {
+    input:
+      tumor_bam = sampleApplyBaseRecalibrator.recalibrated_bam,
+      tumor_bam_index = sampleApplyBaseRecalibrator.recalibrated_bai,
+      normal_bam = normalApplyBaseRecalibrator.recalibrated_bam,
+      normal_bam_index = normalApplyBaseRecalibrator.recalibrated_bai,
+      refGenome = refGenome,
+      genomeReference = af_only_gnomad,
+      genomeReferenceIndex = af_only_gnomad_index
+  }
+
+  call annovar {
+    input:
+      input_vcf = Mutect2.output_vcf,
+      ref_name = refGenome.ref_name,
+      annovarTAR = annovarTAR,
+      annovar_operation = annovar_operation,
+      annovar_protocols = annovar_protocols
+  }
```

Comment:
Similar to the comment above, in reality we will have an array of normal samples. These two tasks need to be run after the normal and tumor loops occur, on a loop that pulls out normal and tumor samples together.

Reply:
Same suggestion as above. Maybe for the next iteration we can show two examples. Also in that case the normal will be processed twice.